src/main/java/org/deepsymmetry/beatlink/data/OpusProvider.java

```diff
@@ -51,72 +59,135 @@ public boolean isRunning() {
+
+        public Database getDatabase() {
+            return database;
+        }
+
+        public FileSystem getFileSystem() {
+            return fileSystem;
+        }
+    }
+
+    /**
+     * Holds the archive information (USB slot number, parsed Database and Zipped Filesystem)
+     * of the 1st media export mounted for one of the Opus USB slots. These are dynamic and do not have
+     * to match Opus USB slot numbers.
      */
-    private final AtomicReference<FileSystem> usb1filesystem = new AtomicReference<>();
+    private final AtomicReference<RekordboxUsbArchive> usb1archive = new AtomicReference<>();
 
     /**
-     * Holds the parsed database for the media we are proxying for the USB 1 slot of the Opus Quad.
+     * Holds the archive information (USB slot number, parsed Database and Zipped Filesystem)
+     * of the 2nd media export mounted for one of the Opus USB slots. These are dynamic and do not have
+     * to match Opus USB slot numbers.
      */
-    private final AtomicReference<Database> usb1database = new AtomicReference<>();
+    private final AtomicReference<RekordboxUsbArchive> usb2archive = new AtomicReference<>();
 
     /**
-     * Holds the ZIP filesystem of the media export mounted for USB slot 2 of the Opus Quad.
+     * Holds the archive information (USB slot number, parsed Database and Zipped Filesystem)
+     * of the 3rd media export mounted for one of the Opus USB slots. These are dynamic and do not have
+     * to match Opus USB slot numbers.
      */
-    private final AtomicReference<FileSystem> usb2filesystem = new AtomicReference<>();
+    private final AtomicReference<RekordboxUsbArchive> usb3archive = new AtomicReference<>();
 
     /**
-     * Holds the parsed database for the media we are proxying for the USB 2 slot of the Opus Quad.
+     * Tracks the Database and FileSystem to use for which Opus player.
      */
-    private final AtomicReference<Database> usb2database = new AtomicReference<>();
+    private final Map<SlotReference, RekordboxUsbArchive> slotReferenceToArchive = new ConcurrentHashMap<>();
+
+    /**
+     * Attach a metadata archive to supply information for the media mounted a USB slot of the Opus Quad.
+     * This is a convenience method to be backwards compatible for BeatLinkTrigger, but this should probably go.
+     *
+     * @param archiveFile the metadata archive that can provide metadata for tracks playing from the specified USB slot, or
+     *                {@code null} to stop providing metadata for that slot
+     * @param slot which TrackSourceSlot slot we are to provide metadata for. They are mapped to the USB slots
+     *
+     * @throws java.io.IOException if there is a problem attaching the archive
+     * @throws IllegalArgumentException if a slot other than the SD or USB slot is specified
+     */
+    @API(status = API.Status.STABLE)
+    public synchronized void attachMetadataArchive(File archiveFile, CdjStatus.TrackSourceSlot slot) throws IOException {
```

Comment:
Convenience method to bridge the codebase for BLT. Should be removed when BLT has caught up.